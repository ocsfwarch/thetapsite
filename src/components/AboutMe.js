import React from "react";

const AboutMe = () => {
  return (
    <div className="aboutme_container">
      <section className="img_sect">
        <section className="img_display">
          <img
            src="/images/AboutMe/tp_image_2.png"
            title="About Me"
            alt="About Me"
          />
        </section>
      </section>
      <section className="intro_sect">
        <p>
          My name is Terry Partridge. I have been a Full Stack Software Engineer
          based in Mission Viejo, California since 2000. For the past 2 years my
          primary focus for frontend and backend development has been the
          PostgreSQL, Express, React, and Node (PERN) stack. I love the creative
          process associated with developing software and the constant advances
          in software development technology. It allows me to build applications
          that help people and it feeds my need for continuous improvement.
        </p>
      </section>
      <section className="footer_sect">
        <section className="app_header_menu">
          <nav className="navContent">
            <section>
              <ul className="navLinks">
                <li>Work History</li>
                <li>Skills</li>
                <li>Education</li>
              </ul>
            </section>
          </nav>
        </section>
      </section>
    </div>
  );
};

export { AboutMe };
