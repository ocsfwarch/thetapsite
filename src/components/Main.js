import React from "react";
import pic from "../images/IMG_9450_small.png";

export const Main = () => {
  return (
    <div className="intro_container">
      <section className="intro_img_container">
        <img src={pic} alt="display" />
        <span className="caption">Grand Canyon AZ from Plateau Point</span>
      </section>
      <section className="intro_content_container">
        <section className="banner">
          <h1>Welcome to my World Wide Web space</h1>
        </section>
        <section className="content">
          <p>
            My name is Terry Partridge. I am a Software Engineer based in
            Mission Viejo, California. This is my space, I hope you find it
            useful.
          </p>
        </section>
      </section>
    </div>
  );
};
