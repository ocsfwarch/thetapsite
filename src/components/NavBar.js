import React from "react";

export const NavBar = () => {
  return (
    <nav className="navContent">
      <section>
        <ul className="navLinks">
          <li>
            <a href="/">Home</a>
          </li>
          <li>
            <a href="/Projects">Projects</a>
          </li>
          <li>
            <a href="/AboutMe">About Me</a>
          </li>
        </ul>
      </section>
    </nav>
  );
};
