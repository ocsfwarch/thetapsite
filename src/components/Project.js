import React from 'react';

export const Project = ({project}) => {
    //console.log(`path=${window.location.pathname}, Image = ${img}`)
    const openProject = () => {
        window.open(project.href, '_blank');
    }

    const techStack = project.stack.map((item,idx) => {
        return(<img key={idx} src={item.src}        title={item.title}      alt={item.alt}></img>)
    });

    return(
        <div className="project_container">
            <section className="project_title">
                <a href={project.href} target="_blank" title="Click to show the app">{project.title}</a>
            </section>
            
            <section className="project_image">
                <section className="project_image_container">
                    <img src={project.image} />
                </section>
            </section>
            <section className="project_summary">
                <p>
                    {project.summary}
                </p>
                <p>
                    {techStack}
                </p>
            </section>

            <section className="project_footer">
                <button type="button" className="btn btn-primary" onClick={openProject}>Visit App</button>
            </section>
        </div>
    );
};
