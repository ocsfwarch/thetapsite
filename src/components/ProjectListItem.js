import React from "react";

export const ProjectListItem = ({ project }) => {
  const techStack = project.stack.map((item, idx) => {
    return (
      <img key={idx} src={item.src} title={item.title} alt={item.alt}></img>
    );
  });

  let appLink = "";
  if (project.href && project.href.length > 0) {
    appLink = (
      <a
        href={project.href}
        target="_blank"
        rel="noreferrer"
        title="Click to show the app"
      >
        <button type="button" className="btn btn-primary">
          Visit App
        </button>
      </a>
    );
  }

  let gitLink = "";
  if (project.git && project.git.length > 0) {
    gitLink = (
      <a
        href={project.git}
        target="_blank"
        rel="noreferrer"
        title="Click to visit the GitHub repository"
      >
        <button type="button" className="btn btn-primary">
          Visit GitHub
        </button>
      </a>
    );
  }

  return (
    <div className="project_container">
      <section className="project_title">
        <a
          href={project.href}
          target="_blank"
          rel="noreferrer"
          title="Click to show the app"
        >
          {project.title}
        </a>
      </section>

      <section className="project_image">
        <img src={project.image} alt="" />
      </section>

      <section className="project_summary">
        <p>{project.summary}</p>
      </section>

      <section className="project_stack">{techStack}</section>

      <section className="project_footer">
        {appLink}
        {gitLink}
      </section>
    </div>
  );
};
