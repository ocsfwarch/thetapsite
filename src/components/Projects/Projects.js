import React, { useState } from "react";
import { ProjectsNav } from "./ProjectsNav";
import { ProjectsList } from "./ProjectsList";

const Projects = () => {
  const [projectType, setProjectType] = useState("enterprise");
  const updateProjectType = (event) => {
    event.preventDefault();
    const newProjectType = event.target.name;
    if (newProjectType && newProjectType.length) {
      if (newProjectType !== projectType) {
        setProjectType(newProjectType);
      }
    }
  };

  return (
    <div className="project_view_container">
      <ProjectsNav
        projectType={projectType}
        updateProjectType={updateProjectType}
      />
      <ProjectsList projectType={projectType} />
    </div>
  );
};

export { Projects };
