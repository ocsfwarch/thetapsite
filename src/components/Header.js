import React from 'react';
import { NavBar } from './NavBar';

export const Header = () => {

    return(
        <div className="app_header">
            <div className="app_header_banner">
                <section ><h1>The TAP Site</h1></section>
            </div>
            <div className={"app_header_menu"}>
                <NavBar />
            </div>
        </div>
    );
};