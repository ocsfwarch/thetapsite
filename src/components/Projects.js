import React from "react";
import { Project } from "./Project";
import { ProjectItems } from "./Helpers/ProjectItems";

export const Projects = () => {
  const projectList = ProjectItems.map((item, idx) => (
    <Project key={idx} project={item} />
  ));
  return <div className="projects_container">{projectList}</div>;
}; // End Projects
