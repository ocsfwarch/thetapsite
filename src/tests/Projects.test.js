import React from "react";
import renderer from "react-test-renderer";
import { Projects } from "../components/Projects";

it("renders correctly", () => {
  const tree = renderer.create(<Projects />).toJSON();
  expect(tree).toMatchSnapshot();
});
