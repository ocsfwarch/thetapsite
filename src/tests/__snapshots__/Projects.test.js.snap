// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="projects_container"
>
  <div
    className="project_container"
  >
    <section
      className="project_title"
    >
      <a
        href="https://www.chatcorenews.com"
        target="_blank"
        title="Click to show the app"
      >
        CHAT Core News
      </a>
    </section>
    <section
      className="project_image"
    >
      <section
        className="project_image_container"
      >
        <img
          src="/images/Projects/chatcorenews_screenshot_desktop.png"
        />
      </section>
    </section>
    <section
      className="project_summary"
    >
      <p>
        This is an RSS feed aggregator service. 
      </p>
      <p>
        <img
          alt="css icon"
          src="/images/Techs/css.png"
          title="CSS 3"
        />
        <img
          alt="html icon"
          src="/images/Techs/html.png"
          title="HTML 5"
        />
        <img
          alt="react icon"
          src="/images/Techs/react.png"
          title="React"
        />
        <img
          alt="js icon"
          src="/images/Techs/js.png"
          title="JavaScript"
        />
        <img
          alt="postgres icon"
          src="/images/Techs/postgresql.png"
          title="PostgreSQL"
        />
        <img
          alt="npm icon"
          src="/images/Techs/npm.png"
          title="npm"
        />
      </p>
    </section>
    <section
      className="project_footer"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Visit App
      </button>
    </section>
  </div>
  <div
    className="project_container"
  >
    <section
      className="project_title"
    >
      <a
        href="https://ocsfwarch.com/ocsatext/"
        target="_blank"
        title="Click to show the app"
      >
        OCSA Text
      </a>
    </section>
    <section
      className="project_image"
    >
      <section
        className="project_image_container"
      >
        <img
          src="/images/Projects/text_screenshot.png"
        />
      </section>
    </section>
    <section
      className="project_summary"
    >
      <p>
        This is a secure text application
      </p>
      <p>
        <img
          alt="css icon"
          src="/images/Techs/css.png"
          title="CSS 3"
        />
        <img
          alt="html icon"
          src="/images/Techs/html.png"
          title="HTML 5"
        />
        <img
          alt="react icon"
          src="/images/Techs/react.png"
          title="React"
        />
        <img
          alt="js icon"
          src="/images/Techs/js.png"
          title="JavaScript"
        />
        <img
          alt="postgres icon"
          src="/images/Techs/postgresql.png"
          title="PostgreSQL"
        />
        <img
          alt="npm icon"
          src="/images/Techs/npm.png"
          title="npm"
        />
      </p>
    </section>
    <section
      className="project_footer"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Visit App
      </button>
    </section>
  </div>
  <div
    className="project_container"
  >
    <section
      className="project_title"
    >
      <a
        href="https://ocsfwarch.com/ocsatracker/"
        target="_blank"
        title="Click to show the app"
      >
        OCSA Tracker
      </a>
    </section>
    <section
      className="project_image"
    >
      <section
        className="project_image_container"
      >
        <img
          src="/images/Projects/tracker_screenshot.png"
        />
      </section>
    </section>
    <section
      className="project_summary"
    >
      <p>
        This is a vehicle maintenance tracking application
      </p>
      <p>
        <img
          alt="css icon"
          src="/images/Techs/css.png"
          title="CSS 3"
        />
        <img
          alt="html icon"
          src="/images/Techs/html.png"
          title="HTML 5"
        />
        <img
          alt="react icon"
          src="/images/Techs/react.png"
          title="React"
        />
        <img
          alt="js icon"
          src="/images/Techs/js.png"
          title="JavaScript"
        />
        <img
          alt="postgres icon"
          src="/images/Techs/postgresql.png"
          title="PostgreSQL"
        />
        <img
          alt="npm icon"
          src="/images/Techs/npm.png"
          title="npm"
        />
      </p>
    </section>
    <section
      className="project_footer"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Visit App
      </button>
    </section>
  </div>
  <div
    className="project_container"
  >
    <section
      className="project_title"
    >
      <a
        href="https://ocsfwarch.com/handyman/"
        target="_blank"
        title="Click to show the app"
      >
        Flagstaff HandyMan Services
      </a>
    </section>
    <section
      className="project_image"
    >
      <section
        className="project_image_container"
      >
        <img
          src="/images/Projects/handyman_screenshot.png"
        />
      </section>
    </section>
    <section
      className="project_summary"
    >
      <p>
        This is an example of a basic services advertising web site.
      </p>
      <p>
        <img
          alt="css icon"
          src="/images/Techs/css.png"
          title="CSS 3"
        />
        <img
          alt="html icon"
          src="/images/Techs/html.png"
          title="HTML 5"
        />
        <img
          alt="react icon"
          src="/images/Techs/react.png"
          title="React"
        />
        <img
          alt="js icon"
          src="/images/Techs/js.png"
          title="JavaScript"
        />
        <img
          alt="postgres icon"
          src="/images/Techs/postgresql.png"
          title="PostgreSQL"
        />
        <img
          alt="npm icon"
          src="/images/Techs/npm.png"
          title="npm"
        />
      </p>
    </section>
    <section
      className="project_footer"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Visit App
      </button>
    </section>
  </div>
</div>
`;
